clash-lib/src/Clash/Primitives/DSL.hs

```diff
@@ -391,12 +391,15 @@ vec els@(el:_)
 
 vec [] = error "vec: can't be used on empty lists"
 
--- | Create a tuple of 'TExpr'
-tuple :: TExpr -> TExpr -> TExpr
-tuple a b =
-  let
-    ty = Product "GHC.Tuple.(,)" Nothing [ety a, ety b]
-  in TExpr ty (DataCon ty (DC (ty,0)) [eex a, eex b])
+-- | Create an n-tuple of 'TExpr'
+tuple :: [TExpr] -> TExpr
+tuple [] = error $ "nTuple: Cannot create empty tuple"
+tuple [_] = error $ "nTuple: Cannot create a 1-tuple"
```

Comment:
Is an error here preferable to a 0-tuple being `()` and a 1-tuple just being the value?

Reply:
> Is an error here preferable to a 0-tuple being ()

Hmm. I'd like to be a bit cautious with `()`. Zero-width paths should almost always be treated with care in blackboxes. Let's add it when we have use case.

> a 1-tuple just being the value?

That seems useful, will add.

Reply:
Fair point about the zero width path